homeassistant/components/forked_daapd/config_flow.py

```diff
@@ -0,0 +1,169 @@
+                            CONF_TTS_PAUSE_TIME, DEFAULT_TTS_PAUSE_TIME
+                        ),
+                    ): float,
+                    vol.Optional(
+                        CONF_TTS_VOLUME,
+                        default=self.config_entry.options.get(
+                            CONF_TTS_VOLUME, DEFAULT_TTS_VOLUME
+                        ),
+                    ): float,
+                    vol.Optional(
+                        CONF_PIPE_CONTROL,
+                        default=self.config_entry.options.get(CONF_PIPE_CONTROL),
+                    ): str,  # https://github.com/home-assistant/core/issues/32819
+                    vol.Optional(
+                        CONF_PIPE_CONTROL_PORT,
+                        default=self.config_entry.options.get(
+                            CONF_PIPE_CONTROL_PORT, DEFAULT_PIPE_CONTROL_PORT
+                        ),
+                    ): int,
+                }
+            ),
+        )
+
+
+def fill_in_schema_dict(some_input):
+    """Fill in schema dict defaults from user_input."""
+    schema_dict = {}
+    for field, _type in DATA_SCHEMA_DICT.items():
+        if some_input.get(str(field)):
+            schema_dict[
+                vol.Optional(str(field), default=some_input[str(field)])
+            ] = _type
+        else:
+            schema_dict[field] = _type
+    return schema_dict
+
+
+class ForkedDaapdFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a forked-daapd config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize."""
+        self.discovery_schema = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Return options flow handler."""
+        return ForkedDaapdOptionsFlowHandler(config_entry)
+
+    async def validate_input(self, user_input):
+        """Validate the user input."""
+        websession = async_get_clientsession(self.hass)
+        return await ForkedDaapdAPI.test_connection(
+            websession=websession,
+            host=user_input[CONF_HOST],
+            port=user_input[CONF_PORT],
+            password=user_input.get(CONF_PASSWORD),
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a forked-daapd config flow start.
+
+        Manage device specific parameters.
+        """
+        if user_input is not None:
+            validate_result = await self.validate_input(user_input)
+            if validate_result == "ok":  # success
+                _LOGGER.debug("Connected successfully. Creating entry.")
+                return self.async_create_entry(
+                    title=f"{FD_NAME} server", data=user_input
+                )
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(fill_in_schema_dict(user_input)),
+                errors={"base": validate_result},
```

Comment:
I understand the need to have an additional decoupler in case the library changes, but for now the decoupler will just be a simple dict that maps the existing strings that the pypi library returns to copies of themselves (since they currently match the strings.json keys). Is that ok?

Reply:
Yes.